### ISSUE TITLE
bug: Kakarot Messaging precompile not applying whitelist

### ISSUE THREAD

**Issue by enitrat** [kkrt-labs/kakarot#1314]
In  `precompiles.cairo::exec_precompile`

```cairo
pre_kakarot_precompile:
let is_cairo_contract_call = Helpers.is_zero(
    FIRST_KAKAROT_PRECOMPILE_ADDRESS - evm_address
);
let is_whitelisted = KakarotPrecompiles.is_caller_whitelisted(caller_address);
tempvar is_authorized = is_cairo_contract_call * is_whitelisted;
tempvar syscall_ptr = syscall_ptr;
tempvar pedersen_ptr = pedersen_ptr;
tempvar range_check_ptr = range_check_ptr;
jmp kakarot_precompile if is_authorized != 0;
jmp unauthorized_call if is_cairo_contract_call != 0;
jmp kakarot_precompile;
```
is_cairo_contract_call will be false, we are calling the  cairo messaging address 0x75002, not the cairo precompile address 0x75001 which is the value of FIRST_KAKAROT_PRECOMPILE_ADDRESS
is_whitelisted will also be false. So will be is_authorized.
Therefore we should not take both conditional jumps and still end up in kakarot_precompile, which will execute the cairo messaging precompile.

Add checks on caller address to make sure only an authorized L1 messaging contract can call Kakarot.

**Comment by obatirou** [comment]
It seems already fixed by PR https://github.com/kkrt-labs/kakarot/pull/1239 which is reflected on main [here](https://github.com/kkrt-labs/kakarot/blob/67fbb3d44f83fd3d2e0c674a2d0a9db65d1033c3/src/kakarot/precompiles/precompiles.cairo#L143)
Now, the call is authorized only if it is a whitelisted address and a cairo_module call

**Comment by enitrat** [comment]
> Now, the call is authorized only if it is a whitelisted address and a cairo_module call

Yes but there is still no check whether a call to KakarotPrecompiles.cairo_message is authorized through `Kakarot_authorized_message_senders` right ?

**Comment by obatirou** [comment]
> > Now, the call is authorized only if it is a whitelisted address and a cairo_module call
> 
> Yes but there is still no check whether a call to KakarotPrecompiles.cairo_message is authorized through `Kakarot_authorized_message_senders` right ?

Indeed, it seems I missed something here.
There is a check with `Kakarot_authorized_cairo_precompiles_callers` but not `Kakarot_authorized_message_senders`
Could you clarify the difference between the 2 ?
My understanding: `Kakarot_authorized_cairo_precompiles_callers` would be evm accounts (Kakarot or L1) that can call precompiles and `Kakarot_authorized_message_senders` would be Ethereum L1 accounts that can call Kakarot ?

If I am not mistaken, why there is a a whitelisting of evm contracts that can call precompiles ?
Also maybe some renaming should be done ?

**Comment by enitrat** [comment]
`Kakarot_authorized_message_senders` should only be the L1 address of the L1KakarotMessaging contract, that pre-formats messages to be compatible with Kakarot.

`Kakarot_authorized_cairo_precompiles_callers` refers to L2 EVM addresses that are authorized to call Cairo contracts. They are two different, distinct sets

**Comment by obatirou** [comment]
Thank you for the explanation of the 2 sets. There is still something not clear to me.
In the `handle_L1_message` in [kakarot contract](https://github.com/kkrt-labs/kakarot/blob/7fb90d799a5a0bfab0a48bdfcb4e867d35b4943c/src/kakarot/kakarot.cairo#L493) there is a check on `Kakarot_authorized_message_senders`. So any calls coming from L1 will be checked against `Kakarot_authorized_message_senders`.
Why would a check in `KakarotPrecompiles.cairo_message` be needed ?

In `KakarotPrecompiles.cairo_message`, Kakarot will be sending a L2 message to L1 if not mistaken 
`send_message_to_l1(target_address, data_bytes_len, data_ptr);` will send the cairo message and the target_address might not be `L1KakarotMessaging`.

Do I miss something here ?

**Comment by enitrat** [comment]
> In the handle_L1_message in [kakarot contract](https://github.com/kkrt-labs/kakarot/blob/7fb90d799a5a0bfab0a48bdfcb4e867d35b4943c/src/kakarot/kakarot.cairo#L493) there is a check on Kakarot_authorized_message_senders. So any calls coming from L1 will be checked against Kakarot_authorized_message_senders.
Why would a check in KakarotPrecompiles.cairo_message be needed ?

This is totally true and I was mistaken :) Sorry, indeed. No check needed in this specific place. Perhaps the variable names are not clear enough or a code documentation was missing.

**Comment by obatirou** [comment]
> > In the handle_L1_message in [kakarot contract](https://github.com/kkrt-labs/kakarot/blob/7fb90d799a5a0bfab0a48bdfcb4e867d35b4943c/src/kakarot/kakarot.cairo#L493) there is a check on Kakarot_authorized_message_senders. So any calls coming from L1 will be checked against Kakarot_authorized_message_senders.
> > Why would a check in KakarotPrecompiles.cairo_message be needed ?
> 
> This is totally true and I was mistaken :) Sorry, indeed. No check needed in this specific place. Perhaps the variable names are not clear enough or a code documentation was missing.

No worries ! I think we can close this issue then ?
Also an issue for renaming will be welcomed to really grasp the intention of those variable
